contracts/arbitration/GTCR.sol

```diff
@@ -0,0 +1,1013 @@
+/* solium-disable max-len*/
+
+pragma solidity ^0.5.10;
+
+import "./Arbitrable.sol";
+import "./CappedMath.sol";
+
+/**
+ *  @title GeneralizedTCR
+ *  This contract is a curated registry for any types of items. Just like TCR contract it uses request-challenge protocol and crowdfunding, but also has new features such as badges and request cancellation.
+ */
+contract GeneralizedTCR is Arbitrable {
+    using CappedMath for uint;
+
+    /* Enums */
+
+    enum Status {
+        Absent, // The item is not in the registry.
+        Registered, // The item is in the registry.
+        RegistrationRequested, // The item has a request to be added to the registry.
+        ClearingRequested // The item has a request to be removed from the registry.
+    }
+
+    enum Party {
+        None, // Party per default when there is no challenger or requester. Also used for unconclusive ruling.
+        Requester, // Party that made the request to change a status.
+        Challenger // Party that challenges the request to change a status.
+    }
+
+    /* Structs */
+
+    struct Item {
+        bytes itemData; // The data describing the item.
```

Comment:
I suggest we shorten the name of this field to just `data` so it is not redundant.